rich/console.py

```diff
@@ -1682,18 +1682,18 @@ def log(
             if style is not None:
                 renderables = [Styled(renderable, style) for renderable in renderables]
 
-            caller = inspect.stack()[_stack_offset]
-            link_path = (
-                None
-                if caller.filename.startswith("<")
-                else os.path.abspath(caller.filename)
-            )
-            path = caller.filename.rpartition(os.sep)[-1]
-            line_no = caller.lineno
+            caller = inspect.currentframe()
```

Comment:
According to the docs this function may return None if the interpreter isn't CPython. Not sure what supports it and what doesn't (Pypy maybe), but there would need to be a fallback that doesn't break.